internal/stacks/stack_manager.go

```diff
@@ -774,21 +791,25 @@ func (s *StackManager) getBlockchainProvider(verbose bool) blockchain.IBlockchai
 	}
 }
 
-func (s *StackManager) getTokensProvider(verbose bool) tokens.ITokensProvider {
-	switch s.Stack.TokensProvider {
-	case NilTokens.String():
-		return &niltokens.NilTokensProvider{
-			Verbose: verbose,
-			Log:     s.Log,
-			Stack:   s.Stack,
-		}
-	case ERC1155.String():
-		return &erc1155.ERC1155Provider{
-			Verbose: verbose,
-			Log:     s.Log,
-			Stack:   s.Stack,
+func (s *StackManager) getITokenProviders(verbose bool) []tokens.ITokensProvider {
+	tps := make([]tokens.ITokensProvider, len(s.Stack.TokenProviders))
+	for i, tp := range s.Stack.TokenProviders {
+		switch tp {
+		case ERC1155:
+			tps[i] = &erc1155.ERC1155Provider{
+				Verbose: verbose,
+				Log:     s.Log,
+				Stack:   s.Stack,
+			}
+		case ERC20_ERC721:
+			tps[i] = &erc20erc721.ERC20ERC721Provider{
+				Verbose: verbose,
+				Log:     s.Log,
+				Stack:   s.Stack,
+			}
+		default:
+			return nil
```

Comment:
So this will ensure that if I pass `-t none` anywhere on the command line (regardless of any other `-t` options), the behavior is that I get no token providers at all - right? I think that behavior is good and explainable, just want to be sure I'm understanding.

Reply:
This is a good point, and should be the way it works. But it isn’t as the code is implemented. There’s no way to turn off all token providers on Ethereum as it is currently, so I’ll sort that before merge.